packages/cli/src/commands/link/android/patches/makeBuildPatch.js

```diff
@@ -18,6 +19,28 @@ export default function makeBuildPatch(name) {
   return {
     installPattern,
     pattern: /[^ \t]dependencies {(\r\n|\n)/,
-    patch: `    implementation project(':${normalizedProjectName}')\n`,
+    patch: makePatchString(normalizedProjectName, buildGradlePath),
   };
 }
+
+function makePatchString(normalizedProjectName, buildGradlePath) {
+  const defaultPatchString = `    implementation project(':${normalizedProjectName}')\n`;
+  if (!buildGradlePath) {
+    return defaultPatchString;
+  }
+
+  const buildGradle = fs.readFileSync(buildGradlePath);
+
+  const depTypes = ['compile', 'api', 'implementation'];
+  for (const type of depTypes) {
+    const depPattern = new RegExp(
+      `(${type})\\w*\\s*\\(*project\\(['"]:${normalizedProjectName}['"]\\)`,
```

Comment:
Should we care about a whitespace between `project` and `(:lib-name)`? It's our CLI that generates it, so that shouldn't be an issue unless somebody modifies gradle config, maybe with a formatter?

Reply:
True, we should account for possible white spaces in that spot, just in case.